hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactionPipeline.java

```diff
@@ -149,6 +175,15 @@ private void swapSuffix(LinkedList<ImmutableSegment> suffix, ImmutableSegment se
     pipeline.addLast(segment);
   }
 
+  /* Replacing one representation of the segment with another. Literally removing a segment and
+  adding a new one (with same keys). The order of the segments is not kept */
+  private void replaceSegment(ImmutableSegment oldSegment, ImmutableSegment newSegment) {
+    version++;
+    // don't call oldSegment.close() because its MSLAB is transferred to the new segment
+    pipeline.remove(oldSegment);
+    pipeline.add(newSegment);
```

Comment:
will this add the new segment at the same location of the removed segment? i don't think so